src/App.jsx

```diff
@@ -14,33 +15,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+export const App = () => {
+  const [data, setData] = useState(goodsFromServer);
+  const [reversedStatus, setReversedStatus] = useState(false);
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  const customSort = (sortParam = 'alphabet') => {
+    return () => {
+      let sortedData = [...data];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+      switch (sortParam) {
+        case 'alphabet':
+          sortedData.sort((a, b) => a.localeCompare(b));
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+          if (reversedStatus === true) {
+            sortedData.reverse();
+          }
+
+          break;
+
+        case 'length':
+          sortedData.sort((a, b) => a.length - b.length);
+
+          if (reversedStatus === true) {
+            sortedData.reverse();
+          }
+
+          break;
+
+        case 'reverse':
+          sortedData.reverse();
+
+          if (reversedStatus === false) {
+            setReversedStatus(true);
+          } else {
+            setReversedStatus(false);
+          }
+
+          break;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        default:
+          sortedData = goodsFromServer;
+          setReversedStatus(false);
+
+          break;
+      }
+
+      setData(sortedData);
+    };
+  };
```

Comment:
This function is quite large and does multiple things depending on the `sortParam` passed to it. This could make it harder to understand and maintain. Consider breaking it up into smaller functions, each handling one specific task (sorting alphabetically, sorting by length, reversing, resetting). This would improve code readability and maintainability.